codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamFrameToHttpObjectCodec.java

```diff
@@ -132,4 +143,44 @@ protected void encode(ChannelHandlerContext ctx, HttpObject obj, List<Object> ou
         }
         ReferenceCountUtil.retain(obj);
     }
+
+    private Http2Headers toHttp2Headers(final HttpMessage msg) {
+        if (msg instanceof HttpRequest) {
+            msg.headers().set(
+                    HttpConversionUtil.ExtensionHeaderNames.SCHEME.text(),
+                    scheme.name());
+        }
+
+        return HttpConversionUtil.toHttp2Headers(msg, validateHeaders);
+    }
+
+    private HttpMessage newMessage(final int id,
+                                   final Http2Headers headers) throws Http2Exception {
+        return isServer ?
+                HttpConversionUtil.toHttpRequest(id, headers, validateHeaders) :
+                HttpConversionUtil.toHttpResponse(id, headers, validateHeaders);
+    }
+
+    private FullHttpMessage newFullMessage(final int id,
+                                           final Http2Headers headers,
+                                           final ByteBufAllocator alloc) throws Http2Exception {
+        return isServer ?
+                HttpConversionUtil.toFullHttpRequest(id, headers, alloc, validateHeaders) :
+                HttpConversionUtil.toFullHttpResponse(id, headers, alloc, validateHeaders);
+    }
+
+    @Override
+    public void handlerAdded(final ChannelHandlerContext ctx) throws Exception {
+        super.handlerAdded(ctx);
+
+        if (isSsl(ctx.channel())) {
```

Comment:
maybe just use:

```java
scheme = isSsl(...) ? HttpScheme.HTTPS : HttpScheme.HTTP;
```

This way its more explicit about what is used and what not

Reply:
yup, changed